Features/BanCommand.cs

```diff
@@ -0,0 +1,89 @@
+using DSharpPlus.Commands;
+using DSharpPlus.Entities;
+using DSharpPlus.Commands.ContextChecks;
+using ArtcordAdminBot.Features.Helpers;
+using System.IO;
+using System.Net.Http;
+using System.Text.Json.Serialization;
+
+namespace ArtcordAdminBot.Features
+{
+    public class BanCommand
+    {
+        private readonly IDatabaseService _databaseService;
+        private readonly HttpClient _httpClient;
+
+        public BanCommand(IDatabaseService databaseService)
+        {
+            _databaseService = databaseService;
+            _httpClient = new HttpClient();
+        }
+
+        [Command("ban")]
+        [RequirePermissions(DiscordPermissions.BanMembers)]
+        public async Task BanAsync(CommandContext ctx, 
+        [System.ComponentModel.Description("The user to ban.")]DiscordUser targetUser, 
+        [System.ComponentModel.Description("The reason for the ban.")] string reason = "No reason provided", 
+        [System.ComponentModel.Description("The attachment of the reference image.")] DiscordAttachment? attachment = null, 
+        [System.ComponentModel.Description("The ID of the reference message.")] ulong? referenceMessageId = null, 
+        [System.ComponentModel.Description("Additional notes about the ban.")]string additionalNotes = "None", 
+        [System.ComponentModel.Description("Whether or not to delete the users messages from the last 7 days")] bool deleteMessages = false)
```

Comment:
Fine for now, but it could instead be a dropdown (create an enum for the duration, and use that instead of bool, then in code turn the eg. `Duration.7Days` into the actual DateTime) of something like an hour, a day, a week or all time.